dozer/cogs/starboard.py

```diff
@@ -0,0 +1,366 @@
+        e.add_field(name="Cancel Emoji", value=config.cancel_emoji)
+        e.add_field(name="Threshold", value=config.threshold)
+        e.set_footer(text=f"For more information, try {ctx.prefix}help starboard")
+        return e
+
+    async def send_to_starboard(self, config, message, reaction_count):
+        """GIven a message which may or may not exist, send it to the starboard"""
+        starboard_channel = message.guild.get_channel(config.channel_id)
+        if starboard_channel is None:
+            return
+
+        db_msgs = await StarboardMessage.get_by(message_id=message.id)
+        if len(db_msgs) == 0:
+            sent_msg = await starboard_channel.send(embed=make_starboard_embed(message, reaction_count))
+            db_msg = StarboardMessage(message.id, message.channel.id, sent_msg.id, message.author.id)
+            await db_msg.update_or_add()
+            await message.add_reaction(config.star_emoji)
+        else:
+            try:
+                sent_msg = await self.bot.get_channel(config.channel_id).fetch_message(db_msgs[0].starboard_message_id)
+            except discord.errors.NotFound:
+                # Uh oh! Starboard message was deleted. Let's try and delete it
+                DOZER_LOGGER.warning(f"Cannot find Starboard Message {db_msgs[0].starboard_message_id} to update")
+                fake_msg = discord.Object(db_msgs[0].starboard_message_id)
+                await self.remove_from_starboard(config, fake_msg, True)
+                return
+            await sent_msg.edit(embed=make_starboard_embed(message, reaction_count-1))
+
+    async def remove_from_starboard(self, config, starboard_message, cancel=False):
+        """Given a starboard message or snowflake, remove that message and remove it from the DB"""
+        db_msgs = await StarboardMessage.get_by(starboard_message_id=starboard_message.id)
+        if len(db_msgs):
+            if hasattr(starboard_message, 'delete'):
+                await starboard_message.delete()
+            if cancel:
+                try:
+                    orig_msg = await self.bot.get_channel(db_msgs[0].channel_id).fetch_message(db_msgs[0].message_id)
+                    await orig_msg.add_reaction(config.cancel_emoji)
+                except discord.NotFound:
+                    pass
+            await StarboardMessage.delete(message_id=db_msgs[0].message_id)
+
+    async def starboard_check(self, reaction, member):
+        """Provides all logic for checking and updating the Starboard"""
+        msg = reaction.message
+        if not msg.guild:
+            return
+
+        if msg.guild.id in self.config_cache:
+            config = self.config_cache[msg.guild]
+        else:
+            config_lst = await StarboardConfig.get_by(guild_id=msg.guild.id)
+            if len(config_lst) == 1:
+                self.config_cache[msg.guild] = config_lst[0]
+                config = config_lst[0]
+            else:
+                return
```

Comment:
This looks like it could be simplified using [`dozer.db.ConfigCache`](https://github.com/FRCDiscord/Dozer/blob/84da5ed91e6a949351a2c68a74f9394ab37b9106/dozer/db.py#L132-L170).